### PR TITLE
Added support for SE and EfficientNet backbones

### PR DESCRIPTION
**Pull request by ZFTurbo**
Added support for following pre-trained backbones:
'seresnet18', 'seresnet34', 'seresnet50', 'seresnet101', 'seresnet152', 'seresnext50', 'seresnext101', 'senet154', 'EfficientNetB0', 'EfficientNetB1', 'EfficientNetB2', 'EfficientNetB3', 'EfficientNetB4', 'EfficientNetB5', 'EfficientNetB6', 'EfficientNetB7'

It has dependency on following python modules: '[classification_models](https://github.com/qubvel/classification_models)' and '[efficientnet](https://github.com/qubvel/efficientnet)'. But they will be called only if user will use these backbones.

### CODE REVIEW

**Comment by ZFTurbo** [comment]
I currently experimenting with these backbones. I was able to get 0.40 mAP with SE-Resnext50 with freezed backbone. Which is worse than best achieved 0.49 mAP for resnet152. Currently I try to investigate why mAP became worse when I'm unfreeze backbone. Freeze of BN layers didn't help.

File: keras_retinanet/models/effnet.py [hunk 1]
```diff
@@ -41,7 +41,15 @@ def retinanet(self, *args, **kwargs):
...
         """
-        return
+        from efficientnet.model import BASE_WEIGHTS_PATH
+        from efficientnet.model import WEIGHTS_HASHES
+
+        model_name = 'efficientnet-b' + self.backbone[-1]
+        file_name = model_name + '_weights_tf_dim_ordering_tf_kernels_autoaugment_notop.h5'
+        file_hash = WEIGHTS_HASHES[model_name][1]
+        weights_path = get_file(file_name, BASE_WEIGHTS_PATH + file_name, cache_subdir='models', file_hash=file_hash)
+        # print('Image net weights loaded: {}'.format(weights_path))
```
Comment (vcarpani) [inline keras_retinanet/models/effnet.py]: Forgot to remove debud 'print' ;)

**Comment by vcarpani** [comment]
> I currently experimenting with these backbones. I was able to get 0.40 mAP on Open Images Dataset with SE-Resnext50 with freezed backbone. Which is worse than best achieved 0.49 mAP for resnet152. Currently I try to investigate why mAP became worse when I'm unfreeze backbone. Freeze of BN layers didn't help.

Do you have any comparison between ResNext 50 and Resnet 50?

**Comment by ZFTurbo** [comment]
I was able to get 0.39 mAP for resnet50 before unfreeze. So SE-ResNext with 0.40 is a little bit better. But overall I got 0.44 mAP for ResNet50 (after unfreeze).

**Comment by vcarpani** [comment]
LGTM, I am merging so also other users can start testing with different backbones ;)

**Pull request merged**